docs: added scaffolding, config, proto and frontend (#891)
tendermint/starport: docs/6 Protocol Buffer Files/1 Overview.md

# Protocol Buffer Files Overview

> Data structures used by Cosmos SDK modules are defined using protocol buffer files. Inside `proto` directory each custom module has a directory containing `query.proto`, `tx.proto`, `genesis.proto` and other files.

Reviewer: ```suggestion
Protocol buffer files define the data structures used by Cosmos SDK modules. 

Inside the `proto` directory, a directory for each custom module contains `query.proto`, `tx.proto`, `genesis.proto`, and other files.
```

> `starport serve` automatically generates Go code from proto files on every file change.
> 
> Third-party proto files, including those of Cosmos SDK and Tendermint are bundled with Starport. You can import them in your custom proto files like so:

Reviewer: ```suggestion
The `starport serve` command automatically generates Go code from proto files on every file change.

Third-party proto files, including those of Cosmos SDK and Tendermint, are bundled with Starport. To import third-party proto files in your custom proto files:
```

```proto
import "cosmos/base/query/v1beta1/pagination.proto";
```

> Additional third-party proto files can be added manually. By default Starport imports proto files from two directories: `third_party/proto` and `proto_vendor`. This can be customised in `config.yml`:

Reviewer: ```suggestion
You can also manually add third-party proto files. By default, Starport imports proto files from these directories: `third_party/proto` and `proto_vendor`. You can define third-party paths of the import directory in `config.yml`:
```
